### DISCUSSION TITLE
Mapping shift-enter and similar in terminal vim

### DISCUSSION THREAD

**Discussion by marlun** [kovidgoyal/kitty#3392] [Q&A] [answered]
In iTerm2 a solution is to map it to for example F13,14,15,etc as can be seen here: https://groups.google.com/g/iterm2-discuss/c/VEiCSc5LCIs/m/U5fqrwNFm88J?pli=1 Is it possible to do the same in Kitty? I'm reading the comments in the preference file for keyboard shortcuts but I'm not sure if and how I can do it?

**Comment by kovidgoyal** [comment, marked as answer]
You can map any key to send any bytes using the send_text directive in kitty.conf.

**Reply by marlun** [reply]
I'm trying with the following

    map shift+0x24 send_text O2P

but vim is not acting on it.

**Reply by marlun** [reply]
I was missing the first argument to send_text, sending "all" fixed it.

Is there more information on what the different args to send_text mean? Like application or all.

**Reply by marlun** [reply]
It also doesn't work in tmux but I'm not sure it did with iTerm either.

**Reply by marlun** [reply]
Sorry for the spamming but for others wondering it does work in neovim but then you have to change the mapping to actually use <F13>:

    inoremap <F13> ...

**Reply by kovidgoyal** [reply]
https://sw.kovidgoyal.net/kitty/conf.html#shortcut-kitty.Send-arbitrary-text-on-key-presses
